### PR TITLE
fix: show duplication error for predefined section

### PR DESCRIPTION
**Pull request by ishche**

### CODE REVIEW

File: server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java [hunk 1]
```diff
@@ -163,6 +167,13 @@ public Optional<SyntaxError> registerParagraphNameNode(ParagraphNameNode node) {
    * @return syntax error if the code block duplicates
    */
   public Optional<SyntaxError> registerSectionNameNode(SectionNameNode node) {
+    if (sectionMap.containsKey(node.getName())) {
```
Comment (Nurkambay) [inline server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java:170]: Maybe here better to create a new method `boolean containSection(String sectionName)` and move this logic to the place where you call this `registerSectionNameNode` method.

Comment (ishche) [reply server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java:170]: Maybe... but not much. fixed.

**Review by Nurkambay — APPROVED** [review]

**Comment by VitGottwald** [comment]
Nice, thank you! This revealed an issue we have not seen yet. The predefined sections should be only included if they do not exist in the program :-)

**Comment by VitGottwald** [comment]
LGTM. Thank you!

**Pull request merged**